api/clients/codecs/coeff_poly.go

```diff
@@ -0,0 +1,63 @@
+package codecs
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/encoding/fft"
+	"github.com/Layr-Labs/eigenda/encoding/rs"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+)
+
+// coeffPoly is a polynomial in coefficient form.
+//
+// The underlying bytes represent 32 byte field elements, and each field element represents a coefficient
+type coeffPoly struct {
+	fieldElements []fr.Element
+}
```

Comment:
This duplicates an existing type (currently in `encoding/rs/frame_coeffs.go` in `master, named something different if you aren't up to date with the latest `master`).

```
type FrameCoeffs []fr.Element
```

Also, there is a bunch of serialization logic in that file as well for going back and forth between serialized and deserialized form.

Is the plan to circle back and replace these old types in a follow up PR?

Reply:
I see `[]fr.Element` used directly as a type in many places in this PR. Is there a reason why you don't use the fancy type you define here?

Reply:
Is there a reason why you put this into a struct instead of doing `type coefPoly []fr.Element`?

Reply:
When I made the suggested change, `type coefPoly []fr.Element`, everything started simplifying down, culminating in the removal of the type entirely.

Reply:
isn't best to keep the type around to have type safety when applying FFT/IFFT operations (can't by mistake apply FFT twice for eg)